pip/compat/__init__.py

```diff
@@ -72,7 +72,11 @@ def console_to_str(s):
         try:
             return s.decode(sys.__stdout__.encoding)
         except UnicodeDecodeError:
-            return s.decode('utf_8')
+            try:
+                return s.decode('utf_8')
+            except UnicodeDecodeError:
+                # Decode at least line breaks manually
+                return repr(s)[2:-1].replace('\\r', '\r').replace('\\n', '\n')
```

Comment:
Thanks @xavfernandez. Do you think what category is the best, RuntimeWarning, UnicodeWarning or no specific category?